v1/python-sdk/tutorials/automl-with-azureml/forecasting-demand-forecasting-many-models/auto-ml-forecasting-demand-forecasting-many-models.ipynb

```diff
@@ -0,0 +1,1626 @@
+    "\n",
+    "Output of inference step can be represented as [OutputFileDatasetConfig](https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.data.output_dataset_config.outputdatasetconfig?view=azure-ml-py) object which, in turn, will be registered as a dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.data import OutputFileDatasetConfig\n",
+    "\n",
+    "output_inference_data_ds = OutputFileDatasetConfig(\n",
+    "    name=\"many_models_inference_output\",\n",
+    "    destination=(datastore, \"uci_electro_small/inference_output/\"),\n",
+    ").register_on_complete(name=\"uci_electro_small_inference_data_ds\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inference_dataset.take(5).to_pandas_dataframe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For many models we need to provide the ManyModelsInferenceParameters object."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.contrib.automl.pipeline.steps import AutoMLPipelineBuilder\n",
+    "from azureml.train.automl.runtime._many_models.many_models_parameters import (\n",
+    "    ManyModelsInferenceParameters,\n",
+    ")\n",
+    "\n",
+    "output_file_name = \"parallel_run_step.csv\"\n",
+    "inference_ds_small = inference_dataset\n",
+    "\n",
+    "mm_parameters = ManyModelsInferenceParameters(\n",
+    "    partition_column_names=time_series_id_column_names,\n",
+    "    time_column_name=time_column_name,\n",
+    ")\n",
+    "\n",
+    "inference_steps = AutoMLPipelineBuilder.get_many_models_batch_inference_steps(\n",
+    "    experiment=experiment,\n",
+    "    inference_data=inference_ds_small,\n",
+    "    node_count=2,\n",
+    "    process_count_per_node=8,\n",
+    "    compute_target=compute_target,\n",
+    "    run_invocation_timeout=300,\n",
+    "    output_datastore=output_inference_data_ds,\n",
+    "    train_run_id=training_run.id,\n",
+    "    train_experiment_name=training_run.experiment.name,\n",
+    "    inference_pipeline_parameters=mm_parameters,\n",
+    "    append_row_file_name=output_file_name,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from azureml.pipeline.core import Pipeline\n",
+    "\n",
+    "inference_pipeline = Pipeline(ws, steps=inference_steps)\n",
+    "inference_run = experiment.submit(inference_pipeline)\n",
+    "inference_run.wait_for_completion(show_output=False)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# from azureml.pipeline.core import Pipeline\n",
```

Comment:
Do we need this cell? If we do, let us add the description in the markdown cell above: "If you have already ran the experiment and want to use the previous results, please use the code below instead of running the experiment again."

Reply:
dead code. Removed.